test/solidity/Periphery/ReceiverStargateV2.t.sol

```diff
@@ -0,0 +1,277 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity 0.8.17;
+
+import { Test, TestBase, Vm, LiFiDiamond, DSTest, ILiFi, LibSwap, LibAllowList, console, InvalidAmount, ERC20, UniswapV2Router02 } from "../utils/TestBase.sol";
+import { OnlyContractOwner, UnAuthorized, ExternalCallFailed } from "src/Errors/GenericErrors.sol";
+
+import { ReceiverStargateV2 } from "lifi/Periphery/ReceiverStargateV2.sol";
+import { stdJson } from "forge-std/Script.sol";
+import { ERC20Proxy } from "lifi/Periphery/ERC20Proxy.sol";
+import { Executor } from "lifi/Periphery/Executor.sol";
+import { OFTComposeMsgCodec } from "lifi/Libraries/OFTComposeMsgCodec.sol";
+import { LibBytes } from "lifi/Libraries/LibBytes.sol";
+
+contract ReceiverStargateV2Test is TestBase {
+    using stdJson for string;
+
+    ReceiverStargateV2 internal receiver;
+    bytes32 guid = bytes32(0);
+    address receiverAddress = USER_RECEIVER;
+
+    address public constant STARGATE_USDC_POOL_MAINNET =
+        0xc026395860Db2d07ee33e05fE50ed7bD583189C7;
+    address public constant ENDPOINT_V2_MAINNET =
+        0x1a44076050125825900e736c501f859c50fE728c;
+    uint256 public constant RECOVER_GAS_VALUE = 100000;
+    address stargateRouter;
+    bytes32 internal transferId;
+    Executor executor;
+    ERC20Proxy erc20Proxy;
+
+    event StargateRouterSet(address indexed router);
+    event ExecutorSet(address indexed executor);
+    event RecoverGasSet(uint256 indexed recoverGas);
+
+    function setUp() public {
+        customBlockNumberForForking = 20024274;
+        initTestBase();
+
+        erc20Proxy = new ERC20Proxy(address(this));
+        executor = new Executor(address(erc20Proxy));
+        receiver = new ReceiverStargateV2(
+            address(this),
+            address(executor),
+            ENDPOINT_V2_MAINNET,
+            RECOVER_GAS_VALUE
+        );
+        vm.label(address(receiver), "ReceiverStargateV2");
+        vm.label(address(executor), "Executor");
+        vm.label(address(erc20Proxy), "ERC20Proxy");
+
+        transferId = keccak256("123");
+    }
+
+    function test_OwnerCanPullToken() public {
+        // send token to receiver
+        vm.startPrank(USER_SENDER);
+        dai.transfer(address(receiver), 1000);
+        vm.stopPrank();
+
+        // pull token
+        vm.startPrank(USER_DIAMOND_OWNER);
+
+        receiver.pullToken(ADDRESS_DAI, payable(USER_RECEIVER), 1000);
+
+        assertEq(1000, dai.balanceOf(USER_RECEIVER));
```

Comment:
I would add a test for native for sanity sake too:

```
vm.startPrank(USER_SENDER);
(bool success, ) = USER_SENDER.call{ value: 1000 }("");
vm.stopPrank();
assertTrue(success);
assertEq(0, address(USER_RECEIVER).balance);

vm.startPrank(USER_DIAMOND_OWNER);
receiver.pullToken(address(0), payable(USER_RECEIVER), 1000);
assertEq(1000, address(USER_RECEIVER).balance);
```

Reply:
added